src/server/pfs/db/driver.go

```diff
@@ -1165,12 +1167,9 @@ func getPrefixes(path string) []string {
 }
 
 func getDiffID(commitID string, path string) string {
-	return fmt.Sprintf("%s:%s", commitID, path)
-}
-
-// the equivalent of above except that commitID is a rethink term
-func getDiffIDFromTerm(commitID gorethink.Term, path string) gorethink.Term {
-	return commitID.Add(":" + path)
+	s := fmt.Sprintf("%s:%s", commitID, path)
+	hash := sha256.Sum256([]byte(s))
```

Comment:
Interesting. Admittedly, the context where I heard this was for bloom filters, which a) use k hash functions, and b) are only used in performance-sensitive contexts.

Mainly, I am curious if we have any kind of versioning for repo formats, though (or if we think we'll add one before we get any big, inflexible users), because if the answer is yes we don't really have to worry if this is exactly the right call or not.

Secondarily: How often do we compute this hash? Is it once per (file, worker), once per (block, worker), once per file, etc? I think it is possible to turn several small hashes into one big hash by using double hashing to generate e.g. 4 distinct 64-bit hashes and then concatenating them.